### PR TITLE
Add useDelayEffect to help add a bit of spinner UX around sign up

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Description

Primary goal of this PR is to add a spinner to the "enter handle" part of the sign up flow that shows up after a 1s delay. Sometimes it takes `validateHandle` a few seconds to cross-list the handle against twitter/insta/chain.

Some improvements coupled with this PR:
- generic useDelayedEffect for this pattern (some state change X ms after some other state change)
- move of desktop email page to functional component
- small style improvements to buttons on sign up flow
- fixed loading spinner on password reset modal

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?

Touches sign up

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.

Locally against staging: web + mobile

### How will this change be monitored?

For features that are critical or could fail silently please describe the monitoring/alerting being added.

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/rj-add-sign-up-spinner

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/rj-add-sign-up-spinner

File: src/hooks/README.md [hunk 1]
```diff
@@ -32,3 +32,6 @@ Wraps the react router `useLocation` hook to allow a function to be called on ro
 
 ### `useScript`
 Loads a script in the DOM given a source url
+
+### `useDelayedEffect`
+Runs an effect, but only
```
Comment (rickyrombo) [inline src/hooks/README.md:37]: "but only..."?

**Review by rickyrombo — APPROVED** [review]
Curious why we wait a second before showing the spinner?

File: src/hooks/README.md (hunk 1 above)
Comment (raymondjacobson) [reply src/hooks/README.md:37]: > Curious why we wait a second before showing the spinner?

Just a UX ask forrest had. no strong opinions, but it is possible that this thing finishes before 1s, depending on how long the handle reservation checks happen

Comment (rickyrombo) [reply src/hooks/README.md:37]: What does it look like in the "in-between" state? Also, should it be disabled to prevent multiple clicks?

Comment (raymondjacobson) [reply src/hooks/README.md:37]: chatted "online"!

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/rj-add-sign-up-spinner

**Pull request merged**